pytensor/tensor/subtensor.py

```diff
@@ -2522,22 +2657,43 @@ def c_code(self, node, name, input_names, output_names, sub):
         """
 
     def c_code_cache_version(self):
-        return (8,)
+        return (9,)
+
+    def _check_runtime_broadcasting(self, node, x, y, idx):
+        if y.ndim > 0:
+            y_pt_bcast = node.inputs[1].broadcastable
+
+            if not y_pt_bcast[0] and y.shape[0] == 1 and y.shape[0] != idx.shape[0]:
```

Comment:
Should you be looking at `type.shape` here? Or is this function only called in the perform method

Type hints on the method signature would help :)

Reply:
This is called in the perform method yes, with real x, y inputs

Reply:
Added type hints